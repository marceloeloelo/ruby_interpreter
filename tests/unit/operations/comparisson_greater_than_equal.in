# int vs int

1 >= 2
2 >= 1
2 >= 2

# double vs double

1.1 >= 2.1
2.1 >= 1.1
2.1 >= 2.1

# int vs double

1 >= 2.1
1.1 >= 2
1 >= 1.0

# string vs string

"foo" >= "bar"
"bar" >= "foo"
"foo" >= "foo"
"" >= "foo"
"" >= ""

# string vs int

"foo" >= 1
2 >= "foo"
2 >= "2"

# string vs double

2.1 >= "foo"
"foo" >= 1.1
"1.1" >= 1.1

# bool vs bool

true >= false
true >= true
false >= true
false >= false

# bool vs int

true >= 2
true >= 1
true >= 0
2 >= true
1 >= true
0 >= true

false >= 2
false >= 1
false >= 0
2 >= false
1 >= false
0 >= false

# bool vs nil

true >= nil
nil >= true

false >= nil
nil >= false

# bool vs string

true >= "foo"
true >= ""
"foo" >= true
"" >= true

false >= "foo"
false >= ""
"foo" >= false
"" >= false

# bool vs double

true >= 2.1
true >= 1.0
true >= 0.0
2.1 >= true
1.0 >= true
0.0 >= true

false >= 2.1
false >= 1.0
false >= 0.0
2.1 >= false
1.0 >= false
0.0 >= false

# nil vs nil

nil >= nil

# nil vs int

2 >= nil
1 >= nil
0 >= nil

nil >= 2
nil >= 1
nil >= 0

# nil vs double

2.1 >= nil
1.0 >= nil
0.0 >= nil

nil >= 2.1
nil >= 1.0
nil >= 0.0

 
