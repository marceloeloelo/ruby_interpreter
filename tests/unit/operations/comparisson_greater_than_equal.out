false
true
true
false
true
true
false
false
true
true
false
true
false
true
TypeError: Fixnum can't be coerced into String
